code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm

```diff
@@ -275,38 +275,36 @@ Difficulty: Very Hard
 	max_n_of_items = 200
 	pixel_y = -4
 	use_power = 0
-	var/duplicate = FALSE
 	var/memory_saved = FALSE
 	var/list/stored_items = list()
-	var/list/blacklist = (/obj/item/weapon/spellbook)
+	var/list/blacklist = typecacheof(list(/obj/item/weapon/spellbook))
 
 /obj/machinery/smartfridge/black_box/accept_check(obj/item/O)
-	if(O.type in blacklist)
-		return
-	if(istype(O, /obj/item))
-		return 1
-	return 0
+	if(!istype(O))
+		return FALSE
+	if(is_type_in_typecache(O, blacklist))
+		return FALSE
+	return TRUE
 
 /obj/machinery/smartfridge/black_box/New()
-	..()
-	for(var/obj/machinery/smartfridge/black_box/B in machines)
-		if(B != src)
-			duplicate = 1
-			qdel(src)
+	var/static/obj/machinery/smartfridge/black_box/current = src
+	if (current && current != src)
```

Comment:
So here is a bit of magic some might not know about, static proc vars, they work exactly how you expect them to, the = part only runs once, and the var remains around between calls.